distribution/push_v2.go

```diff
@@ -670,10 +668,9 @@ func numOfMatchingPathComponents(pth string, matchComponents []string) int {
 }
 
 func getPathComponents(path string) []string {
-	// make sure to add docker.io/ prefix to the path
-	named, err := reference.ParseNamed(path)
+	named, err := reference.ParseNormalizedNamed(path)
```

Comment:
This is also called by `numOfMatchingPathComponents`, but this could probably be refactored to be more efficient, possibly just guarantee that the normalized string is passed in.